### PR TITLE
tvheadend update

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  
- Update TVH to latest git available
- Bump ffmpeg package version with added `libtheora` to further enhance usage of ffmpeg's shared librairies to accelerate tvheadend build process.

_Linked issues:_  PR #3992

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### TVH Changes
- Update to latest commit as of June 10th 2020
- Update `libhdhomerun` to latest release of 20200521
- Enable VAAPI support for Intel platforms
- Enable `pngquant` support
- Enable `libtheora` support
- Enable `fdk-acc` support
- Enable re-using most needed shared libraries from FFMPEG
### FFMPEG Changes
- Enable `libtheora` support to allow TVH to reuse all sub libraries from ffmpeg
- Fix `SVT-AV1` and `SVT-HEVC` download package to align with upstream

_Fixed issues:_
- Build failure with pngquant: https://tvheadend.org/issues/5894

### CODE REVIEW

**Comment by th0ma7** [comment]
This is an updated PR superseeding #3992 due to, again, rebasing issues.
Also added full re-using of ffmpeg other media shared libraries.

**Comment by th0ma7** [comment]
@ymartin59 I think I was able to solve the linkeage with ffmpeg shared libraries in order to expedite the build process.  Comments appreciated.

File: cross/libimagequant/Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+PKG_NAME = libimagequant
+PKG_VERS = 2.12.6
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/ImageOptim/libimagequant/archive
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
```
Comment (ymartin59) [inline cross/libimagequant/Makefile:6]: Please set `PKG_DIST_FILE` to avoid "version-only" file in distrib/ and digest

Comment (th0ma7) [reply cross/libimagequant/Makefile:6]: done.

**Comment by th0ma7** [comment]
@ymartin59 finally had time to complete a full re-build of `ffmpeg` + `tvheadend` against it.
Running smooth at home on my `apollolake` arch.
Ready for merge.

**Comment by th0ma7** [comment]
@ymartin59 merging this would allow me to easily migrate `svt-hevc` and `svt-av1` to using cmake framework.

**Comment by th0ma7** [comment]
@ymartin59 I think I finally managed to do a proper rebasing this time!  woot!

As this wasn't yet merged and there where updates available to `libhdhomerun` and new memory fixes for `tvheadend` and took the opportunity to push theses last bits in.

Compiled for all arches successfully and running smooth on my `apollolake` against already merged to master `ffmpeg` changes for 4.2.3.

**Comment by th0ma7** [comment]
> LGTM

Thnx @hgy59 !  now merged.

**Comment by hgy59** [comment]
@th0ma7 unfortunately the builds of this PR failed.
There are download errors.
Since we do want to build all affected packages in the github build action, we do not stop (and report) errors there.

I have just enhanced the action with a Build Status that reports packages that failed to build and fails itself when build errors occured.

extracts of build log (you can download the full logs in this repository under Actions)
```
2020-06-11T12:41:19.9168709Z ===>  Downloading files for expat
2020-06-11T12:41:19.9228156Z wget http://downloads.sourceforge.net/project/expat/expat/2.2.7/expat-2.2.7.tar.bz2
2020-06-11T12:41:30.7644797Z GnuTLS: Error in the pull function.
2020-06-11T12:41:30.7645088Z Unable to establish SSL connection.
2020-06-11T12:41:30.7653456Z make[5]: *** [../../mk/spksrc.download.mk:57: download_target] Error 4
2020-06-11T12:41:30.7670797Z make[4]: *** [../../mk/spksrc.depend.mk:44: depend_target] Error 2
2020-06-11T12:41:30.7670988Z make[3]: *** [../../mk/spksrc.depend.mk:44: depend_target] Error 2
2020-06-11T12:41:30.7671145Z make[2]: *** [../../mk/spksrc.depend.mk:44: depend_target] Error 2
2020-06-11T12:41:30.7671302Z make[1]: *** [../../mk/spksrc.depend.mk:44: depend_target] Error 2
2020-06-11T12:41:30.7671648Z make: *** [../../mk/spksrc.depend.mk:44: depend_target] Error 2
2020-06-11T12:41:30.7672406Z make[5]: Leaving directory '/github/workspace/cross/expat'
```

```
2020-06-11T14:05:15.9626969Z WARNING: none or old ffmpeg libraries were detected
2020-06-11T14:05:15.9627701Z          * use --disable-libav or --enable-ffmpeg_static
2020-06-11T14:05:15.9637655Z ERROR: ffmpeg development support not found
2020-06-11T14:05:15.9642841Z make[1]: *** [../../mk/spksrc.configure.mk:50: configure_target] Error 1
2020-06-11T14:05:15.9643671Z make[1]: Leaving directory '/github/workspace/cross/tvheadend'
2020-06-11T14:05:15.9649497Z make: *** [../../mk/spksrc.depend.mk:44: depend_target] Error 2
2020-06-11T14:05:15.9649981Z make: Leaving directory '/github/workspace/spk/tvheadend'
```

**Comment by th0ma7** [comment]
@hgy59 isn't this just a download issue where next try will simply work at downloading the file?  I had no issues doing a `make all-supported` ?!?

**Comment by ymartin59** [comment]
@hgy59 @th0ma7 Please mark PRs with label "status/to-publish" if there are packages to upload in repository. I have to stack them as repository upload is (still) broken

**Pull request merged**